src/mappings/stateChannel.ts

```diff
@@ -24,30 +24,32 @@ export async function handleChannelOpen(
   const {
     channelId,
     indexer,
-    consumer: creator,
+    consumer: _consumer,
     total,
     price,
     expiredAt,
     deploymentId,
     callback,
   } = event.args;
-
-  const abi = ethers.utils.defaultAbiCoder;
-  const consumer = abi.decode(['address'], callback)[0] as string;
-  const agent = creator === consumer ? consumer : undefined;
+  let consumer = _consumer;
+  let agent: string | undefined = undefined;
+  try {
+    consumer = utils.defaultAbiCoder.decode(['address'], callback)[0] as string;
```

Comment:
The consumer is the `creator` of the channel in the event, the `address` decode from the callback is the real consumer (user), in that case, `agent` should equal `agent = creator !== consumer ? creator : undefined`.

Reply:
if the channel is not created by contract, you won't be able to decode address from callback, it is '0x00..'. In that case, we need to use try-catch and assign consumer = creater in the catch block.

Reply:
I do find an issue with my code, have fixed.

Reply:
Yes, that's correct, so the changes will look like this:
```
  let consumer = undefined;
  let agent: string | undefined = undefined;
  try {
    consumer = utils.defaultAbiCoder.decode(['address'], callback)[0] as string;
    agent = _consumer !== consumer ? _consumer : undefined.
  catch {
    consumer = _consumer;
  }
```

Reply:
@nz-neooo could you confirm that whether the `address` will always includes in the `callback` or only for hosting plan.

Reply:
@mzxyz callback is work for all `ConsumerHost` created channel.